pkg/codegen/go/gen.go

```diff
@@ -64,10 +64,14 @@ type typeDetails struct {
 
 // Title converts the input string to a title case
 // where only the initial letter is upper-cased.
+// It also removes $-prefix if any.
```

Comment:
It would be good to add a note that this doesn't attempt to depuplicate the result (e.g. this will generate bad code for a type with `$ref` and `ref`).

Reply:
Good point! Should we replace `$` with some ugly string to minimize the chance of a clash? Or are we okay to wait until a duplicate actually happens?

Reply:
I think we're okay to wait, I'd just add a note to the comment.